src/java.base/share/classes/java/lang/ProcessBuilder.java

```diff
@@ -1120,6 +1132,20 @@ private Process start(Redirect[] redirects) throws IOException {
                 event.pid = process.pid();
                 event.commit();
             }
+            try {
+                if (LOGGER != null) {
+                    RuntimeException stackTraceEx = new RuntimeException("ProcessBuilder.start() debug");
+                    LOGGER.log(System.Logger.Level.DEBUG, "ProcessBuilder.start(): " +
+                                    Arrays.toString(cmdarray) + ", pid: " + process.pid(),
+                                    stackTraceEx);
+                }
+            } catch (Throwable logEx) {
+                try {
+                    System.err.println("Logging failed: " + logEx.getMessage() +
+                            ", ProcessBuilder.start(): " + Arrays.toString(cmdarray) +
+                            ", pid: " + process.pid());
+                } catch (Throwable thEx) {/* ignore */}
+            }
```

Comment:
Is it really useful to try & catch a potential exception here, or wouldn't be better to let it propagate to the caller? An exception here would indicate a bug somewhere either in the logging backend or in the logging configuration - or possibly OOM or any other kind of VM error and maybe it would be better to just fail in that case, and let the VM exit (unless that exception is caught down the road by the caller?)

I understand why we would not want an exception in logging to prevent System.exit() from exiting, but surely the same constraint doesn't hold for `ProcessBuilder::start`?

Reply:
Since adding logging injects additional code during start(), the idea was to handle exceptions to avoid breaking existing code.
And since the logger is cached during the static initialization of the ProcessBuilder class, a mis-configured logger would cause ExceptionInInitializer, exposing the cause early, though in a pretty ugly way. I may need to reconsider that.